### DISCUSSION TITLE
No serial data from JK-BMS to Arduino nano

### DISCUSSION THREAD

**Discussion by paulsteigel** [syssi/esphome-jk-bms#378] [Q&A]
Hi Syssi,
Thanks for your great project. I tried connecting the arduino nano today to the GND/RX/TX/VBAT port on JK-BMS and tried to log data arriving into D4 (TX), D5 (RX) but I see no data arriving into the arduino nano.
Did I do anything wrong, I doubt the nano would be 5V logic while JK-BMS is 3.2V. Would this be the problem.?
Thank you for your attention!

**Comment by syssi** [comment]
The measurements / status frame must be requested. Listening only isn't sufficient.

**Reply by paulsteigel** [reply]
Hi Sebastian,
Thank you for your response. I use Uno (power via my laptop USB) to hook up the JK-BMS GPS port with a simple softwareserial listener and what I received from Uno serial monitoring is some data (see photo). My serial listener interface use only pin 6 (RX),7 (TX). I unplug pin 7 and still receive some different data from pin 6 (seemed that JK-BMS auto send data via serial).
![image](https://github.com/syssi/esphome-jk-bms/assets/8839597/f0cd5730-5b3f-4dc3-a174-2d2cbb1b2908)
I have consult with my friend and he said that I need to also wire GND of Uno with the GND on the JK-BMS port.
Please be so kind to advice me on this case with thanks!
Ngoc

**Reply by syssi** [reply]
You have to connect TX, RX and GND. At the moment you are receiving noise/garbage because of the missing ground connection. There should be silence as soon all connections are in place.

**Reply by paulsteigel** [reply]
Many thanks to Sebastian, I finally have the device worked!
